src/traces/box/defaults.js

```diff
@@ -109,6 +109,11 @@ function handleSampleDefaults(traceIn, traceOut, coerce, layout) {
     var yLen = yDims && Lib.minRowLength(y);
     var xLen = xDims && Lib.minRowLength(x);
 
+    var calendar = 'gregorian'; // TODO: should we use ax.calendar here?
+    var opts = {
+        autotypenumbers: 'convert types' // TODO: should we use ax.autotypenumbers here?
```

Comment:
oy, I guess in principle yes these should come from `ax` but we haven't yet even created `_fullLayout` axes when we're doing trace defaults, and I don't see a reference to the input layout from here (the `layout` arg is `_fullLayout`), to allow you to pre-construct the axis values. I suppose we could use `layout.calendar` and `layout.autotypenumbers` and that would deal with most cases, just not when the individual axes override those. Seems like a reasonable compromise to me, given the extremely limited cases where this would matter.

Reply:
Thanks. Done in https://github.com/plotly/plotly.js/pull/5240/commits/99a201509782fadc9eaf009d42e68e1b15b7fd2a.

Reply:
Also `box` test fixed by https://github.com/plotly/plotly.js/pull/5240/commits/6fdf5591b93880066fb3c255d5f269cba1a1b8be.